modules/binary-authorization/README.md

```diff
@@ -0,0 +1,57 @@
+# Binary Authorization Infrastructure
+
+This module creates the infrastructure and Attestors necessary to generate attestations on image digests.
+
+## Compatibility/Requirements
+
+* GCP Project ID where the project has an active billing account associated with it
+* Terraform version 0.12+
+* Google Kubernetes Engine cluster with "Binary Authorization" enabled
+
+## Usage
+
+```terraform
+# Create a Key Ring
+resource "google_kms_key_ring" "keyring" {
+  name     = "my-example-attestor-key-ring"
+  location = var.keyring-region
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+# Create Quality Assurance attestor
+module "quality-attestor" {
+  source = "terraform-google-modules/kubernetes-engine/google//modules/binary-authorization"
+
+  attestor-name = "quality-assurance"
+  keyring-id    = google_kms_key_ring.keyring.id
+}
+
+```
+
+## Next Steps
+
+After building the Attestors, Attestations can be associated with image digests.
+
+This module does not include a Binary Authorization policy for a cluster.  A sample policy implemented as Dry-Run/Log-Only using our "quality-assurance" Attestor could look like this:
+
+```terraform
+resource "google_binary_authorization_policy" "policy" {
+
+  admission_whitelist_patterns {
+    name_pattern = "gcr.io/${var.project_id}/*" # Enable local project GCR
+  }
+
+  global_policy_evaluation_mode = "ENABLE"
+
+  # Production ready (all attestors required)
+  default_admission_rule {
+    evaluation_mode  = "REQUIRE_ATTESTATION"
+    enforcement_mode = "DRYRUN_AUDIT_LOG_ONLY"
+    require_attestations_by = [
+      module.quality-attestor.attestor  # Our Attestor
+    ]
+  }
+}
+```
```

Comment:
We should add docs hooks like in other modules ([example](https://raw.githubusercontent.com/terraform-google-modules/terraform-google-kubernetes-engine/fc031fcb693321ec9c54d07d7bf20e6e964fff25/modules/acm/README.md)) for generating the inputs and outputs table.

You can append
```
<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
 <!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
to this file and then run `make docker_generate_docs ` to populate them.

[More details here]( https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/CONTRIBUTING.md#autogeneration-of-documentation-from-tf-files)

Reply:
This is a module that runs alongside the cluster.  The cluster needs to have binary authorization add-on installed.  I had this module in a downstream project and was asked by @morgante the module here, so I could then use the module in the downstream project.

I have run the `make build` on the module after putting in the pre-commit hook lines.

I can also create an `/example/binary-authorization-cluster` if you'd like.  My downstream project has it functional so it'd be easy to extract that from the project.

Reply:
I absolutely agree BinAuthz is a great addition, especially alongside modules like safer cluster. IMHO an example would be great if its not too much trouble, just so that its easier for end users as a starting point.